x-pack/plugins/enterprise_search/common/types/indices.ts

```diff
@@ -28,3 +30,14 @@ export interface ElasticsearchIndex {
   };
   uuid?: Uuid;
 }
+
+export interface ElasticsearchIndexWithIngestion extends ElasticsearchIndex {
```

Comment:
I wonder if its worth specifically enforcing at type level that it should be either/or but not both

Reply:
Tried but failed :'(